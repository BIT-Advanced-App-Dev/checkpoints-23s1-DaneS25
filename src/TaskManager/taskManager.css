@import url('https://fonts.googleapis.com/css2?family=Lora:ital,wght@0,400..700;1,400..700&display=swap');

.taskManager > header{
    font-size: 80px;
    font-family: "Lora", serif;
    text-align: center;
  }
  
  .taskManager__container {
    width: 95%;
    margin: 10px auto 0 auto;
    display: flex;
    justify-content: center;
    flex-direction: column;
  }
  
  .taskManager__container > button {
    font-family: "Lora", serif;
    border-radius: 5px;
    border: none;
    color: white;
    font-size: 18px;
    padding: 10px 15px;
    margin: 0 auto;
    background-color: rgb(125, 188, 125);
    cursor: pointer;
  }
  
  .taskManager__container > button:hover{
    background-color: hsl(117, 86%, 35%);
  }
  
  .taskManager__tasks{
    display: flex;
    flex-direction: column;
    gap: 20px;
    margin-top: 20px;
  }

.logout_btn {
  font-family: "Lora", serif;
  border-radius: 5px;
  border: none;
  color: white;
  font-size: 18px;
  padding: 10px 15px;
  margin: 0 auto;
  background-color: rgb(59, 119, 248);
  cursor: pointer;
  position: relative;
  float: right;
  margin-right: 70px;
  margin-top: -60px;
}

.userDetails {
  font-family: "Lora", serif;
  position: flex;
  float: right;
  margin-right: 20px;
  margin-top: -10px;
}