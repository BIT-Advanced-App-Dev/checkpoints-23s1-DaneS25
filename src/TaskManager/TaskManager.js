/* 
TaskManager component.
Checks what data is stored in the firestore database
Renders said data into the app
*/

import './taskManager.css'
import Task from '../Task/Task'
import {useState, useEffect} from 'react'
import {collection, query, orderBy, onSnapshot} from "firebase/firestore"
import {db} from '../firebase'
import AddTask from '../AddTask/AddTask'

function TaskManager() {

  const [openAddModal, setOpenAddModal] = useState(false) // Open set to false
  const [tasks, setTasks] = useState([])

  // function to get all tasks from firestore in realtime 
  useEffect(() => {
    const taskColRef = query(collection(db, 'tasks'), orderBy('created', 'desc')) // Firestore query to get all tasks and order by created date in descending order
    const unsubscribe = onSnapshot(taskColRef, (snapshot) => { // Listen for changes to Firestore task collection
      setTasks(snapshot.docs.map(doc => ({ // Update tasks state variable with data from each Firestore document
        id: doc.id,
        data: doc.data()
      })))
    })
    return () => {
      unsubscribe() // Unsubscribe from the onSnapshot listener when the component unmounts
    }
  },[])

  return (
    <div className='taskManager'>
      <header>Task Manager</header>
      <div className='taskManager__container'>
        <button 
          onClick={() => setOpenAddModal(true)}>
          Add task +
        </button>
        <div className='taskManager__tasks'>

          {/* Map over all tasks and create a Task component for each one */}
          {tasks.map((task) => (
            <Task
              id={task.id}
              key={task.id}
              completed={task.data.completed}
              title={task.data.title} 
              description={task.data.description}
            />
          ))}

        </div>
      </div>

      {/* Render the AddTask component if the 'openAddModal' state variable is true */}
      {openAddModal &&
        <AddTask onClose={() => setOpenAddModal(false)} open={openAddModal}/>
      }

    </div>
  )
}

export default TaskManager