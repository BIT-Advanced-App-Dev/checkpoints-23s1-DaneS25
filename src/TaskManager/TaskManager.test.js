import React from 'react';
import { render } from '@testing-library/react';
import TaskManager from './TaskManager';

describe('TaskManager', () => {
  it('renders correctly', () => {
    const { container } = render(<TaskManager />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
