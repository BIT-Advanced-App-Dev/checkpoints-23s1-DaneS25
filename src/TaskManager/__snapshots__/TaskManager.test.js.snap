// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskManager renders correctly 1`] = `
<div
  class="taskManager"
>
  <header>
    Task Manager
  </header>
  <div
    class="taskManager__container"
  >
    <button
      name="Add task +"
    >
      Add task +
    </button>
    <div
      class="taskManager__tasks"
    />
  </div>
</div>
`;
